openadr-client/src/lib.rs

```diff
@@ -482,4 +503,136 @@ impl Client {
             program,
         ))
     }
+
+    /// Create a new event on the VTN
+    pub async fn create_event(&self, event_data: EventContent) -> Result<EventClient> {
+        let event = self.client_ref.post("events", &event_data, &[]).await?;
+        Ok(EventClient::from_event(self.client_ref.clone(), event))
+    }
+
+    /// Get a list of events from the VTN with the given query parameters
+    pub async fn get_events_request(
+        &self,
+        target_type: Option<TargetLabel>,
+        targets: &[&str],
+        pagination: PaginationOptions,
+    ) -> Result<Vec<EventClient>> {
+        // convert query params
+        let target_type_str = target_type.map(|t| t.to_string());
+        let skip_str = pagination.skip.to_string();
+        let limit_str = pagination.limit.to_string();
+
+        // insert into query params
+        let mut query = vec![];
+        if let Some(target_type_ref) = &target_type_str {
+            for target in targets {
+                query.push(("targetValues", *target));
+            }
+            query.push(("targetType", target_type_ref.as_str()));
+        }
+        query.push(("skip", &skip_str));
+        query.push(("limit", &limit_str));
+
+        // send request and return response
+        let events: Vec<Event> = self.client_ref.get("events", &query).await?;
+        Ok(events
+            .into_iter()
+            .map(|event| EventClient::from_event(self.client_ref.clone(), event))
+            .collect())
+    }
+
+    /// Get a single event from the VTN that matches the given target
+    pub async fn get_event(&self, target: Target<'_>) -> Result<EventClient> {
+        let pagination = PaginationOptions { skip: 0, limit: 2 };
+
+        let mut events = self
+            .get_events_request(
+                Some(target.target_label()),
+                target.target_values(),
+                pagination,
+            )
+            .await?;
+
+        match events[..] {
+            [] => Err(crate::Error::ObjectNotFound),
+            [_] => Ok(events.remove(0)),
+            [..] => Err(crate::Error::DuplicateObject),
+        }
+    }
```

Comment:
it appears that, even though it is not enforced, a program is sort of expected to not have two events with the same name

Reply:
Might be, however I don't think this will be a common operation. Since for example a timeline for a program will usually have multiple events, so getting one specific event seems to be less useful than getting a single program.

Reply:
so, remove it? that means we remove test coverage too. we cannot make these functions private, because with how we set up the tests, they are in a different crate and cannot use private symbols. 

we might want to change that test setup though? or was that a deliberate choice?

Reply:
or move to `ProgramClient` maybe?

Reply:
Can't we use `get_events_request` to hit the same lines?